### ISSUE TITLE
Midpoint Review - Accessibility Feedback - MHV to VA.gov Team - Medical Records (Comprehensive Domains)

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#57260]
## VFS actions
- [ ] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] Close the ticket when all feedback has been addressed.

## Thoughts/questions

- Just want to reiterate that all of this is tremendous work on a huge product. Moving from health history to allergies to lab results etc is a very consistent and intentional experience, which is tough to pull off.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Should:** I have reservations about the “Change your status” button that dynamically reveals content on the Share your medical record page. Conditionally-revealed content like that is hard to make fully accessible, especially when it also has a form for processing user feedback and notifying the user that a change has occurred. This feature should link to a separate page with that form, which then returns the user to the Share your medical record page and provides a confirmation message about changes made (keyboard focus going to the confirmation message).
   - **Conditional Must:** If that's not practical and the workflow needs to stay on this page, then my must feedback is to test the heck out of it with an accessibility specialist, especially with multiple screen reader/browser combinations to make sure everything is clearly communicated in the interaction.
- **Must:** If the dropdown button is the route you decide to go for the print/download interaction, then the new component needs to go through experimental design. It sounds like you're already looking at the [W3C example pattern](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-actions-active-descendant/), which is probably the best starting point. But W3C examples are sometimes theoretical ideals and browser vendors sometimes interpret standards differently than expected. My main source of anxiety here is that `aria-haspopup` only has [partial support with Android TalkBack](https://a11ysupport.io/tech/aria/aria-haspopup_attribute#support-table-6). This is something to test with an accessibility specialist before you get too committed to using this pattern. (Preferably an accessibility specialist with an Android device!)
   - **Must:** Menu items in the dropdown should have at least a 44px by 44px click/touch target. This is coming from [WCAG 2.1 SC 2.5.5](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html). Strictly speaking, the VA.gov experience standard that references 2.5.5 only says it's for buttons, but I think that's mostly because we didn't really anticipate links being used in this kind of context (there are exceptions in the WCAG standard for inline links). Regardless of link/button semantics, a 44x44 target size will help avoid accidental mis-clicks and mis-taps.
   - **Should:** Maintain the material honesty of the print/download actions in the dropdown menu. Elsewhere on VA.gov, printing is an action triggered by a `button`, while downloading a file is often (but not exclusively) a link. For voice command software users in particular, the distinction between a link and a button is important for how they interact with the page, so the visual presentation should communicate those semantics to the user. Since this is a new component that will go through experimental design you have some flexibility in establishing some new visual styling norms within the dropdown. I don't have specific design recommendations here, except that buttons and links shouldn't have identical styling in this context.
   - **Consider:** Maybe consider using something like the additional info component or a re-styled accordion component to collapse the print/download actions rather than creating a new component. This sidesteps the `aria-haspopup` issue and may provide more space to resolving the material honest issue above.
- **Consider:** When viewing test result details, is the "What's the standard range?" additional info component going to contain the same information each time? If so, consider if there's a way to only include it once on a page. For screen reader users tabbing through the page or having their screen reader announce a list of links/buttons, getting repeated "What's the standard range?" is a little annoying. If there's a way you can get away with only using it once, that would be a cleaner experience.
   - **Conditional must:** If the standard range additional info has different information for some or all of the test types, then the triggering element will need to have unique text each time, eg. "What's the standard range for [test type]?" That will help screen reader users distinguish between the several standard range additional info's and decide which one to open. In the prototype they just contain generic information about standard ranges, so this is only if the actual data contain something specific for each type of test.
- **Consider:** Cheryl flagged this as a content "must" in #57219 and referenced a screen reader issue with the mixed-case on the email address `VAmhvfeedback@va.gov`. The capitalization might lead to some odd pronunciation by screen readers. Truthfully, even with any capitalization/non-capitalization approach `vamhvfeedback` might be pronounced oddly. Screen reader users will be able to manually have their computer speak each letter out loud if they need it, but I think you might get better results on the first pass with all lower-case letters. I don't know if you have the ability to change it but you'll probably get the best results with an email address that doesn't have abbreviations in it at all!
- **Engineering note:** Each “Details” link and "View full image" link will need a unique accessible name. That can be either…
   - Unique link text that's visible to all users (`<a href=”#”>Details about back pain from April 1, 2022</a>`)
   - An `aria-label` to provide unique text to screen reader users (`<a href=”#” aria-label=”Details about back pain from April 1, 2022”>Details</a>`).
   
   If possible, I would prefer that unique text to be visible to all users! If that’s not practical for design purposes, the `aria-label` is a must.
- **Engineering note:** In situations where it's possible for headings to be repeated multiple times on a page (eg. a user's medical history includes multiple instances of back pain), screen reader users will benefit from some way to distinguish them when navigating the page by heading. I believe using `aria-describedby` to associate the heading with its date will be your best bet here, ie. something like `<h2 aria-describedby="some-unique-id">Back pain</h2><p id="some-unique-id">April 1, 2022</p>`.
- **Engineering note:** For pagination, make sure keyboard focus behavior when moving from page to page matches the behavior that currently exists on the VA.gov sitewide search.

**Comment by BobbyBaileyRB** [comment]
- **Should: I have reservations about the “Change your status” button that dynamically reveals content on the Share your medical record page.
Conditionally-revealed content like that is hard to make fully accessible, especially when it also has a form for processing user feedback and notifying the user that a change has occurred. This feature should link to a separate page with that form, which then returns the user to the Share your medical record page and provides a confirmation message about changes made (keyboard focus going to the confirmation message).
**Conditional Must**: If that's not practical and the workflow needs to stay on this page, then my must feedback is to test the heck out of it with an accessibility specialist, especially with multiple screen reader/browser combinations to make sure everything is clearly communicated in the interaction.

There is a component built on the VA staging website, found within the My Profile page. It is the Gender identity option. A Veteran can interact with the edit button then it opens a list of radio buttons to select from. Angela Fowler tested this button and interaction for accessibility – she reported back saying “The gender identity option component handles it very well.” This experience is the same as the “Change your status” option. As of now, we are continuing with this experience and will follow how the Gender identity component was built. We will continue to test it and improve any accessibility concerns. If needed, if this component has not gone through an experimental design, then we can submit a request for it.

- **Must: If the dropdown button is the route you decide to go for the print/download interaction, then the new component needs to go through experimental design. It sounds like you're already looking at the W3C example pattern, which is probably the best starting point. But W3C examples are sometimes theoretical ideals and browser vendors sometimes interpret standards differently than expected. My main source of anxiety here is that aria-haspopup only has partial support with Android TalkBack. This is something to test with an accessibility specialist before you get too committed to using this pattern. (Preferably an accessibility specialist with an Android device!)**

We are working closely with accessibility on this component and asking questions to ensure the best route toward accessibility and usability. Our next steps are to submit an experimental design for the dropdown button.

- **Must: Menu items in the dropdown should have at least a 44px by 44px click/touch target. This is coming from WCAG 2.1 SC 2.5.5. Strictly speaking, the VA.gov experience standard that references 2.5.5 only says it's for buttons, but I think that's mostly because we didn't really anticipate links being used in this kind of context (there are exceptions in the WCAG standard for inline links). Regardless of link/button semantics, a 44x44 target size will help avoid accidental mis-clicks and mis-taps.**

In our designs, we have updated the dropdown menu items (buttons) to have a 44px by 44px click/touch target.
